Sources/SocksCore/TCPSocket.swift

```diff
@@ -203,12 +203,15 @@ public class TCPInternetSocket: InternetSocket, TCPSocket, TCPReadableSocket, TC
     public func close() throws {
         if closed { return }
         stopWatching()
+        closed = true
         if socket_close(self.descriptor) != 0 {
+            if errno != EBADF {
```

Comment:
So basically, closed should be considered true unless it's a bad descriptor, EBADF? I think I'd prefer to set once instead of toggle incase there's some type of observing behavior at one point, maybe something like this

```swift
guard socket_close(self.descriptor) == 0 else {
    // bad file descriptor EBADF should still be considered closed ... include reference here
    if errno == EBADF { closed = true }
    throw SocksError(.closeSocketFailed)
}

// set descriptor to -1 to prevent further use
self.descriptor = -1
closed = true
```

Which brings up another question for me, can we get these two things

1. a link to docs that show why `EBADF` is still considered closed 
2. on `EBADF`, we're considering the socket, but not closing the descriptor w/ -1 and we're throwing. Should it still throw on this error, and should descriptor be set to -1 as well?